src/io.c

```diff
@@ -125,14 +109,46 @@ void memory_write(memory_t *m, uint32_t addr, const uint8_t *src, uint32_t size)
         uint32_t x = p >> 16;
         chunk_t *c = m->chunks[x];
         if (!c) {
-            c = malloc(sizeof(chunk_t));
-            memset(c->data, 0, sizeof(c->data));
+            c = calloc(1, sizeof(chunk_t));
             m->chunks[x] = c;
         }
         c->data[p & 0xffff] = src[i];
     }
 }
 
+void memory_write_w(memory_t *m, uint32_t addr, const uint8_t *src)
+{
+    const uint32_t addr_lo = addr & mask_lo, addr_hi = addr >> 16;
+    chunk_t *c = m->chunks[addr_hi];
+    if (unlikely(!c)) {
+        c = calloc(1, sizeof(chunk_t));
```

Comment:
Can you eliminate the use of `calloc` here?

Reply:
If we want to eliminate it, we should know the total memory size and allocate it first.

Reply:
Let's introduce memory pool within the pull request.

Reply:
You don't have to integrate TLSF (or its variants such as xvmalloc) within this commit. Instead, an arbitrary memory pool works for this case.

Reply:
I roughly integrate the [TLSF 2.4.6](http://www.gii.upv.es/tlsf/main/repo) to file `io.c`.